plantStatsContract/src/main/java/com/example/plantstatscontract/PlantStatisticsContract.kt

```diff
@@ -1,15 +1,92 @@
 package com.example.plantstatscontract
 
+import com.example.plantsapp.domain.model.Plant
+import com.example.plantsapp.domain.model.Task
+import com.example.plantsapp.domain.model.TaskKeys
+import java.util.Date
+
 object PlantStatisticsContract {
-    private const val AUTHORITY =
+    const val AUTHORITY =
         "com.example.plantsapp.PlantsContentProvider"
-    private const val PLANTS_PATH = "plants"
-    const val CONTENT_TYPE =
-        "vnd.android.cursor.dir/vnd.$AUTHORITY.$PLANTS_PATH"
-    const val CONTENT_URI =
-        "content://$AUTHORITY/$PLANTS_PATH"
-
-    const val FIELD_PLANT_NAME = "FIELD_PLANT_NAME"
-    const val FIELD_SPECIES_NAME = "FIELD_SPECIES_NAME"
-    const val FIELD_PLANT_PICTURE = "FIELD_PLANT_PICTURE"
+
+    object Plants {
+        const val PATH = "plants"
+        const val CONTENT_TYPE = "vnd.android.cursor.dir/vnd.$AUTHORITY.$PATH"
+        const val CONTENT_URI = "content://$AUTHORITY/$PATH"
+
+        const val FIELD_NAME = "FIELD_PLANT_NAME"
+        const val FIELD_SPECIES_NAME = "FIELD_SPECIES_NAME"
+        const val FIELD_PICTURE = "FIELD_PLANT_PICTURE"
+    }
+
+    object Tasks {
+        const val PATH = "tasks"
+        const val CONTENT_TYPE = "vnd.android.cursor.dir/vnd.$AUTHORITY.$PATH"
+        const val CONTENT_URI = "content://$AUTHORITY/$PATH"
+
+        const val FIELD_TASK_KEY = "FIELD_TASK_KEY"
+        const val FIELD_FREQUENCY = "FIELD_TASK_FREQUENCY"
+        const val FIELD_LAST_UPDATE_DATE = "FIELD_LAST_UPDATE_DATE"
+    }
+
+    object TaskHistory {
+        const val PATH = "taskHistory"
+        const val CONTENT_TYPE = "vnd.android.cursor.dir/vnd.$AUTHORITY.$PATH"
+        const val CONTENT_URI = "content://$AUTHORITY/$PATH"
+
+        const val FIELD_COMPLETION_DATE = "FIELD_TASK_COMPLETION_DATE"
+    }
+
+    object SelectionArgs {
+        private const val ARGUMENT_PLANT_NAME = 0
+        private const val ARGUMENT_PLANT_SPECIES_NAME = 1
+        private const val ARGUMENT_PLANT_PICTURE = 2
+
+        private const val ARGUMENT_TASK_KEY = 3
+        private const val ARGUMENT_TASK_FREQUENCY = 4
+        private const val ARGUMENT_TASK_LAST_UPDATE_DATE = 5
+
+        fun putPlantInArgs(plant: Plant): Array<out String?> {
```

Comment:
Is it needed to use `out`?
If yes, then ok and you can merge
If not - remove and then merge